pkg/controller/manager/manager_controller.go

```diff
@@ -640,6 +643,17 @@ func (r *ReconcileManager) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, err
 	}
 
+	// Check if non-cluster host log ingestion is enabled.
+	// non-cluster hosts will only forward logs to a standalone or a managed cluster.
+	var nonclusterhost *operatorv1.NonClusterHost
+	if managementCluster == nil {
```

Comment:
I was wondering if there is any harm in removing this if-statement.

Reply:
If a nonclusterhost resource is accidentally applied to a management cluster, this check prevents opening the ingestion endpoint.

Reply:
I meant to say that I don't really see the problem in having a management cluster that also has non cluster hosts working at the same time.

Reply:
At a minimum, I would prefer if we inverted this check - i.e., "If both are specified, error" instead of silently ignoring an unsupported configuration.

I'm fine with erroring for now if MCM is out of scope, but also fine to allow it in management clusters if we think it will come relatively for free.

Reply:
`spec.endpoint` is validated in the manager controller. When `NonClusterHost` resource is applied to the cluster and the `endpoint` filed is an invalid URL, manager controller will be degraded.

The `managementCluster == nil` check is also removed because management cluster should support this feature.